src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -111,6 +128,7 @@ message RunTaskResponse {
     optional bool is_reducer = 6;
     optional bool success = 7;
     optional Metrics metrics = 8;
+    optional FailureInfo failure = 9;  // Implies !success
```

Comment:
Let's stop using success here and say that if this is not None then the task failed.

Reply:
+1 -- once these land and are deployed we should remove the success field.